File: python/contains_duplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # return len(set(nums)) < len(nums)
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False

File: go/contains_duplicate/contains_duplicate.go
package kata

// ::KATA START::
func containsDuplicate(nums []int) bool {
	hashSet := map[int]struct{}{}
	for _, num := range nums {
		if _, exists := hashSet[num]; exists {
			return true
		}
		hashSet[num] = struct{}{}
	}

	return false 
}
// ::KATA END::
